components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -98,6 +103,13 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
             authenticatedUser.setUserResidentOrganization(authorizedUser.getUserResidentOrganization());
         }
 
+        UserAssociation userAssociation = fetchUserAssociation(authenticatedUser, accessingOrgId);
+        /* After sharing all the admin users to the organizations via an migration, this if check should block switching
+        if user association not exists. */
+        if (userAssociation != null) {
+            authenticatedUser.setUserId(userAssociation.getUserId());
```

Comment:
@thanujalk @AnuradhaSK 
Here, the userID of the user who is available in the accessing/switching organization is set. By setting this at this place, make easier for the scope validators as no any addition logic to be written to resolve the shared user.

As this is a token request flow, where no user activity is logged, setting the userID of the sharedUser should not be a problem. Also userID is not kept in DB. It is always resolved by domain qualified user name when necessary. 

This kind of logic should not be implemented in resource access flow, as those events are required to track which user did which action. Please comment on this.

Reply:
this can lead to future bugs. Can't we resolve the user when needed?

Reply:
Yes, it would be better. Will track for the moment till discuss and get approved from @SujanSanjula96 to merge this PR

Reply:
This logic is reverted - https://github.com/wso2-extensions/identity-oauth2-grant-organization-switch/pull/22